internal/nexodus/wg_peers.go

```diff
@@ -268,21 +269,7 @@ func (nx *Nexodus) buildPeersConfig() map[string]client.ModelsDevice {
 
 			if nx.nexRelay.myDerp != CustomDerpRegionID {
 				nx.nexRelay.SetCustomDERPMap(derpAddr, hostname)
-				cmManual := nx.certModeManual(relayDevice)
-
-				if cmManual {
-					// If the cert mode is manual, we need to set the local DNS entry
-					// to point it to the provided hostname
-					ip, _, err := net.SplitHostPort(derpAddr)
-					if err != nil {
-						nx.logger.Errorf("Failed to parse stun address %s : %v", derpAddr, err)
-					}
-
-					err = SetLocalDerpDnsEntry(ip, hostname)
```

Comment:
@chirino I don't see this being set anywhere else (context/dnscache), so wondering how does it will resolve the dns request, we don't explicitly set it?

Reply:
See: https://github.com/nexodus-io/nexodus/pull/1979/files#diff-7b699c477aec224ffe7d58ab48015134c0c67b55615982ccc78c62cfbdec0553R300-R303

We just set them using the extra hosts option when we spin up the test containers.

Reply:
yeah I saw that, but thats covers the test part. But we do need this when we deploy this in cluster. If user onboard it's own private derp relay using it's own certificates (--certmode manual), the dns won't be resolved by public relay, so currently we are using configuring that dns entry in the /etc/host. Sorry if you are already doing that in the PR and i missed it.

Reply:
We can totally ditch setting up entry in /etc/host, and asked user to do it by itself as well, but because we have a better visibility to reflexive address on the onboarded relay, it's makes things bit convenient for user, and they don't have to run the stun client etc to figure out the address.

Reply:
As you suggested on slack, i think we can use http dialer context to resolve the dns to the reflexive ip address of the onboarded derp relay, but unfortunately most of the connect related call is in derp package.

Reply:
I am just wondering if we can hook our own dns resolved to the derp http client 
https://github.com/nexodus-io/nexodus/blob/main/pkg/derp/derphttp/derphttp_client.go#L55

Reply:
Yeah, I think I'm fine with letting users setup DNS themselves for now.  Using the stun discovered reflective IP is not a sure way to get the IP:port of the derp proxy so I think we should not do that as the primary resolution method.  If we could fallback to that (when the user did not setup DNS), then maybe that would be ok.

Reply:
I updated the PR to do that.  We now set a fallback resolver that uses the reflexive IP of the derp relay.  So we should still have the previous functionality.  Sadly, the current e2e tests can't exercise it since we can't depend on using the reflexive IP on OSX.